### ISSUE TITLE
Can you ELI5 how husky works?

### ISSUE THREAD

**Issue by ripper234** [typicode/husky#331]
Let's say Joe sets up a fresh project, adds husky and a precommit hook, then pushes.

Now Alice pulls the repo. She then just edits a file and commits (without running `npm install`). When does husky even get a chance to run and install the precommit hooks? git after all doesn't parse `package.json` files and doesn't care about husky...

So when in this process is husky run? What happens when husky rules are modified, how does it sync itself to git precommit hooks?

**Comment by alexeyten** [comment]
In described process it wouldn't run.

**Comment by ripper234** [comment]
Right.

So let's say Alice does `npm install`. I guess this actually installs both precommit & prepush hooks into git that in turn do `npm run precommit` and `npm run prepush`?

**Comment by typicode** [comment]
> So let's say Alice does npm install. I guess this actually installs both precommit & prepush hooks into git that in turn do npm run precommit and npm run prepush?

@ripper234 Correct :+1: `npm install` needs to be run so that hooks are installed :)

Thanks @alexeyten for the help.
